gamemode/sv_bannedmodels.lua

```diff
@@ -1,102 +1,110 @@
-GM.BannedModels = {}
+-- This file is what controls what models are banned. Models that are banned
+-- cannot be chosen as a disguise.
+
+
+GM.BannedModels = {} -- This is used as a hash table where the key is the model string and the value is true.
+
+
+util.AddNetworkString("ph_bannedmodels_getall")
+util.AddNetworkString("ph_bannedmodels_add")
+util.AddNetworkString("ph_bannedmodels_remove")
 
-util.AddNetworkString("ph_bannedmodels")
 
 function GM:IsModelBanned(model)
-	return table.HasValue(self.BannedModels, model)
+	return self.BannedModels[model] == true
 end
 
+
 function GM:AddBannedModel(model)
-	table.insert(self.BannedModels, model)
+	if self.BannedModels[model] == true then return end -- Prevent duplicates.
```

Comment:
More of this set stuff you don't need to deduplicated

Also, this seems duplicated between the different files. Should this all just be in a `sh_bannedmodels.lua` or something? Probably not a big deal.

Reply:
This duplicate check is definitely not as important as in cl_bannedmodels because having a duplicate model here just means we're doing an unnecessary SaveBannedModels and writing to disk. However, I think this is fine to stay since the check is probably faster than saving the file.